Source/Libraries/GSF.PhasorProtocols/SelCWS/RollingPhaseEstimationTest.cs

```diff
@@ -0,0 +1,250 @@
+
+        const int TotalSamples = (int)(sampleRate / nominalFreq * 5);
+        const long SamplePeriodNs = (long)(1e9 / sampleRate);
+        const double Omega = 2.0 * Math.PI * nominalFreq;
+
+        long epochNs = 0;
+
+        PhaseEstimate lastEstimate = new();
+        bool gotEstimate = false;
+
+        for (int i = 0; i < TotalSamples; i++)
+        {
+            double t = i / sampleRate;
+
+            double ia = peakAmplitudes[0] * Math.Sin(Omega * t);
+            double ib = peakAmplitudes[1] * Math.Sin(Omega * t - 2.0 * Math.PI / 3.0);
+            double ic = peakAmplitudes[2] * Math.Sin(Omega * t + 2.0 * Math.PI / 3.0);
+            double va = peakAmplitudes[3] * Math.Sin(Omega * t);
+            double vb = peakAmplitudes[4] * Math.Sin(Omega * t - 2.0 * Math.PI / 3.0);
+            double vc = peakAmplitudes[5] * Math.Sin(Omega * t + 2.0 * Math.PI / 3.0);
+
+            if (estimator.Step(ia, ib, ic, va, vb, vc, epochNs, out PhaseEstimate? estimate))
+            {
+                lastEstimate = estimate.Value;
+                gotEstimate = true;
+            }
+
+            epochNs += SamplePeriodNs;
+        }
+
+        if (gotEstimate)
+        {
+            string[] names = { "IA", "IB", "IC", "VA", "VB", "VC" };
+            for (int i = 0; i < 6; i++)
+            {
+                double error = Math.Abs(lastEstimate.Magnitudes[i] - expectedRms[i]);
+                double errorPct = error / expectedRms[i] * 100;
+                System.Console.WriteLine($"  {names[i]}: Measured={lastEstimate.Magnitudes[i]:F4}, Expected={expectedRms[i]:F4}, Error={errorPct:F2}%");
+            }
+        }
+        System.Console.WriteLine();
+    }
+
+    static void TestPhaseAngles()
+    {
+        System.Console.WriteLine("Test 4: Phase Angle Relationships");
+        System.Console.WriteLine("----------------------------------");
+
+        const double sampleRate = 3000.0;
+        const double nominalFreq = 60.0;
+        const double amplitude = 100.0;
+
+        RollingPhaseEstimator estimator = new(sampleRate, LineFrequency.Hz60);
+
+        // Currents lead voltages by 30 degrees (capacitive load)
+        const double CurrentLeadRad = 30.0 * Math.PI / 180.0;
+        const int TotalSamples = (int)(sampleRate / nominalFreq * 5);
+        const long SamplePeriodNs = (long)(1e9 / sampleRate);
+        long epochNs = 0;
+
+        PhaseEstimate lastEstimate = new();
+        bool gotEstimate = false;
+
+        for (int i = 0; i < TotalSamples; i++)
+        {
+            double t = i / sampleRate;
+            double omega = 2.0 * Math.PI * nominalFreq;
+
+            // Currents with 30° lead
+            double ia = amplitude * Math.Sin(omega * t + CurrentLeadRad);
+            double ib = amplitude * Math.Sin(omega * t - 2.0 * Math.PI / 3.0 + CurrentLeadRad);
+            double ic = amplitude * Math.Sin(omega * t + 2.0 * Math.PI / 3.0 + CurrentLeadRad);
```

Comment:
If you _add_ 30 degrees to the phase shift, doesn't that mean current is lagging? I suppose the test still works this way, but it looks backwards.